src/System.IO.Pipes/tests/PipeTest.Write.cs

```diff
@@ -133,6 +133,23 @@ public virtual void ReadOnWriteOnlyPipe_Throws_NotSupportedException()
         }
 
         [Fact]
+        public async Task WriteZeroLengthBuffer_Nop()
+        {
+            using (ServerClientPair pair = CreateServerClientPair())
+            {
+                PipeStream pipe = pair.writeablePipe;
+
+                // Shouldn't throw
+                pipe.Write(Array.Empty<byte>(), 0, 0);
+
+                Task writeTask = pipe.WriteAsync(Array.Empty<byte>(), 0, 0);
+                await writeTask;
+
+                Assert.Equal(TaskStatus.RanToCompletion, writeTask.Status);
```

Comment:
This assert is superfluous.  If writeTask ended as anything other than RanToCompletion, we'll never get to this assert because the await on writeTask will throw.